CA2022 errors

New SDK introduces new warnings that turn into errors in VMR build - see https://github.com/dotnet/installer/pull/19145#issuecomment-2027844450

```
/vmr/src/nuget-client/src/NuGet.Core/NuGet.Packaging/Signing/Archive/SignedPackageArchiveUtility.cs(107,13): error CA2022: Avoid inexact read with 'System.IO.Stream.Read(byte[], int, int)' (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2022) [/vmr/src/nuget-client/src/NuGet.Core/NuGet.Packaging/NuGet.Packaging.csproj::TargetFramework=net9.0]
/vmr/src/nuget-client/src/NuGet.Core/NuGet.ProjectModel/Utf8JsonStreamReader.cs(55,13): error CA2022: Avoid inexact read with 'System.IO.Stream.Read(byte[], int, int)' (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2022) [/vmr/src/nuget-client/src/NuGet.Core/NuGet.ProjectModel/NuGet.ProjectModel.csproj::TargetFramework=net9.0]
```

I'm going to patch to disable this warning, in VMR builds. However this needs to be properly fixed.

--- a/src/NuGet.Core/NuGet.Packaging/Signing/Archive/SignedPackageArchiveUtility.cs
+++ b/src/NuGet.Core/NuGet.Packaging/Signing/Archive/SignedPackageArchiveUtility.cs
@@ -104,7 +104,9 @@ namespace NuGet.Packaging.Signing
             var buffer = new byte[localFileHeader.UncompressedSize];
 
             reader.BaseStream.Seek(offsetToData, SeekOrigin.Begin);
+#pragma warning disable CA2022 // Avoid inexact read
             reader.BaseStream.Read(buffer, offset: 0, count: buffer.Length);
+#pragma warning restore CA2022
 
             return new MemoryStream(buffer, writable: false);
         }

--- a/src/NuGet.Core/NuGet.ProjectModel/Utf8JsonStreamReader.cs
+++ b/src/NuGet.Core/NuGet.ProjectModel/Utf8JsonStreamReader.cs
@@ -52,7 +52,9 @@ namespace NuGet.ProjectModel
             _buffer = _bufferPool.Rent(bufferSize);
             _disposed = false;
             _stream = stream;
+#pragma warning disable CA2022 // Avoid inexact read
             _stream.Read(_buffer, 0, 3);
+#pragma warning restore CA2022
             if (!Utf8Bom.AsSpan().SequenceEqual(_buffer.AsSpan(0, 3)))
             {
                 _bufferUsed = 3;

